Fix #5729: Add filter lists to iOS

cuba opened the pull request:
<!-- *Thank you for submitting a pull request, your contributions are greatly appreciated!* -->

## Summary of Changes

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #5729

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
<!-- Any useful notes explaining how best to test and verify. -->

## Links:
Security review: https://github.com/brave/security/issues/975

## Screenshots:
<!-- If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here. -->
![Simulator Screen Shot - iPhone 13 Pro - 2022-07-22 at 10 40 12](https://user-images.githubusercontent.com/909331/180485199-03194c21-d17c-4ec8-aa16-f42004d82b9a.png)

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

cuba committed:
Fix #5729: Add filter lists screen and manager

iccub reviewed (commented):
There's few todos left, make sure they are addressed, if not, create followup tickets for them and refer to them in code

stoletheminerals reviewed (approved):
lgtm from the security pov. Connections to the server where filters are stored aren't pinned, but it's a separate issue https://github.com/brave/brave-ios/issues/5789

cuba committed:
Remove TODOs

iccub reviewed (changes requested):
1. Fix tests, CI fails
2. Recreate new core data model to pick up changes we merged into development recently

I ran this branch for a bit and functionality wise all seems to be working

cuba committed:
Made improvements based on comments

kylehickinson reviewed (changes requested):
- Debug prints should probably use the standard logger or OSLog instead of `print`
- We should add `Sendable` conformances to new classes, or `@unchecked Sendable` if we're certain something is thread-safe when `Sendable` can't be added
- Probably want to add `MainActor` annotations to CoreData functions that require main thread access but I'll open a separate issue for that

cuba committed:
Cleanup for review

cuba committed:
More cleanup for review

iccub reviewed (approved):
My part of review approved

cuba committed:
Fix for review

cuba committed:
Remove useless checkCancellation

cuba committed:
Increase timeouts of ResourceDownloaderStreamTests

iccub committed:
Tone down logs, disable OS_ACTIVITY_MODE

iccub committed:
Kyle Review p2: Sendable, actors, code rework

- ContentBlockerManager sendable.
- Simplify 1 method per Kyle's suggestion.
- Experiment with moving to actors for easier Sendable conformity.

iccub committed:
Add unhandled Sendable for the future.

iccub committed:
Remove CAPS from custom lists header.
